src/creds/auth.rs

```diff
@@ -0,0 +1,52 @@
+use crate::{crypto::crypto_service::Encryptor, logger::logger_manager::Logger};
+
+use super::cred_manager::{CredsManager, ENCRYPT_KEY, IV_PATTERN};
+
+pub trait Authenticator {
+    fn authenticate(&self, username: &str, password: &str) -> bool;
+}
+
+impl Authenticator for CredsManager {
+    fn authenticate(&self, username: &str, password: &str) -> bool {
+        let encryptor = Encryptor::new(ENCRYPT_KEY, IV_PATTERN);
+
+        if let Some(user) = self.users.get(username) {
+            match encryptor.decrypt(&user.password) {
+                Some(decrypted_password) => {
+                    if decrypted_password == password {
+                        return true;
+                    } else {
+                        if self.enable_log == true {
+                            let message =
+                                &format!("password not match with user : {:?}", username.clone());
+                            let set_log = Logger::log_warn_data(message);
+                            set_log.write_log_to_file();
+                        }
+                        println!("Passwords do not match for user: {}", username);
+                        return false;
```

Comment:
this is dupe code in this file, please make a function for this.

Reply:
its better to do this later we have lot of work  to do now for the project.
I will create a new issue for cleanup code

Reply:
it is so major!!!

Reply:
i made issue for this we can handle it later
in issue #44